internal/storage/v2/clickhouse/clickhousetest/server.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package clickhousetest
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/ClickHouse/ch-go/proto"
+	"github.com/ClickHouse/clickhouse-go/v2"
+	chproto "github.com/ClickHouse/clickhouse-go/v2/lib/proto"
+)
+
+var (
+	PingQuery      = "SELECT 1"
+	HandshakeQuery = "SELECT displayName(), version(), revision(), timezone()"
+)
+
+type FailureConfig map[string]error
+
+// NewServer creates a new HTTP test server that simulates a ClickHouse server.
+// It should only be used in tests.
+func NewServer(failures FailureConfig) *httptest.Server {
```

Comment:
you probably need tests for this? or is the coverage derived from other packages?

Reply:
@yurishkuro This function is only called from tests so its covered. Do you prefer to have tests for it?